source/extensions/filters/http/jwt_authn/authenticator.cc

```diff
@@ -130,6 +129,14 @@ void AuthenticatorImpl::verify(Http::HeaderMap& headers, Tracing::Span& parent_s
     return;
   }
 
+  // If token found in cache skip startVerify()
+  if (jwks_data_->getTokenCache()->lookupTokenCache(tokens_.back()->token(), *jwt_)) {
+    ENVOY_LOG(debug, "{}: verified token with cache hit: tokens size {}", name(), tokens_.size());
+    doneWithStatus(Status::Ok);
```

Comment:
Since you are calling doneWithStatus(), please make sure this jwt_ will not get add into cache again.

Reply:
Before adding to cache checking. if present, will not be added to the cache.